### PR TITLE
*: add data type default values

### PR DESCRIPTION
**Pull request by zimulala**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
This feature issue: https://github.com/pingcap/tidb/issues/50936

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: data-type-default-values.md [hunk 1]
```diff
@@ -24,3 +24,37 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+
+`BLOB`、`TEXT` 以及 `JSON` 数据类型写成表达式时，才能为其分配默认值：
+
+```sql
+CREATE TABLE t2 (b BLOB DEFAULT (rand()));
+```
+
+TiDB 目前开始支持一些表达式，具体如下：
+
+* RAND, UUID, UUID_TO_BIN
```
Comment (zimulala) [inline data-type-default-values.md:44]: 这几个表达式函数本身就存在，我不确定是否添加到这里（我没有额外添加测试）。

File: data-type-default-values.md [hunk 2]
```diff
@@ -24,3 +24,37 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+
+## 表达式默认值
+
+> **警告：**
+>
+> 此功能目前为实验特性。
+
+在 `DEFAULT` 子句中指定的默认值可以是表达式。此功能从 MySQL 8.0.13 开始引入的，具体可参考 [data-type-defaults-explicit](https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html#data-type-defaults-explicit)。
+
+`BLOB`、`TEXT` 以及 `JSON` 数据类型写成表达式时，才能为其分配默认值：
```
Comment (Frank945946) [inline data-type-default-values.md:36]:
```suggestion
从 8.0.0 版本开始，`BLOB`、`TEXT` 以及 `JSON` 数据类型支持为其分配默认值，但是默认值仅支持通过表达式来设置。
```

File: data-type-default-values.md [hunk 3]
```diff
@@ -24,3 +24,37 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+
+```sql
+CREATE TABLE t2 (b BLOB DEFAULT (rand()));
+```
+
+TiDB 目前开始支持一些表达式，具体如下：
+
+* RAND, UUID, UUID_TO_BIN
+
```
Comment (Frank945946) [inline data-type-default-values.md:45]:
```suggestion
从 8.0.0 版本开始，`DEFAULT` 子句中支持用以下表达式来设置默认值。
```

**Review by Frank945946 — APPROVED** [review]

File: data-type-default-values.md [hunk 4]
```diff
@@ -24,3 +24,37 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
 * 对于数值类型，它们的默认值是 0。当有 `AUTO_INCREMENT` 参数时，默认值会按照增量情况赋予正确的值。
 * 对于除了时间戳外的日期时间类型，默认值会是该类型的“零值”。时间戳类型的默认值会是当前的时间。
 * 对于除枚举以外的字符串类型，默认值会是空字符串。对于枚举类型，默认值是枚举中的第一个值。
+
+## 表达式默认值
+
+> **警告：**
+>
+> 此功能目前为实验特性。
```
Comment (hfxsd) [inline data-type-default-values.md:32]:
```suggestion
> **警告：**
>
> 该特性为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```

File: data-type-default-values.md [hunk 5]
```diff
@@ -24,3 +24,37 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+  * DATE_FORMAT(NOW(),'%Y-%m')
+  * DATE_FORMAT(NOW(),'%Y-%m-%d')
+  * DATE_FORMAT(NOW(),'%Y-%m-%d %H.%i.%s')
+  * DATE_FORMAT(NOW(),'%Y-%m-%d %H:%i:%s')
+
+* STR_TO_DATE('1980-01-01','%Y-%m-%d')
+
+> **注意：**
+>
+> 目前不支持 `modify/change column` 语句
```
Comment (hfxsd) [inline data-type-default-values.md:60]:
```suggestion
> 目前不支持 `CHANGE COLUMN` 和 `MODIFY COLUMN `语句。
```

File: data-type-default-values.md [hunk 6]
```diff
@@ -24,3 +24,37 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+  * DATE_FORMAT(NOW(),'%Y-%m')
+  * DATE_FORMAT(NOW(),'%Y-%m-%d')
+  * DATE_FORMAT(NOW(),'%Y-%m-%d %H.%i.%s')
+  * DATE_FORMAT(NOW(),'%Y-%m-%d %H:%i:%s')
+
+* STR_TO_DATE('1980-01-01','%Y-%m-%d')
+
+> **注意：**
+>
+> 目前不支持 `CHANGE COLUMN` 和 `MODIFY COLUMN `语句。
```
Comment (hfxsd) [inline data-type-default-values.md:60]:
```suggestion
> 目前不支持 `CHANGE COLUMN` 和 `MODIFY COLUMN` 语句。
```

File: data-type-default-values.md [hunk 7]
```diff
@@ -24,3 +24,37 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+
+* REPLACE(UPPER(uuid()), '-', '')
+
+* DATE_FORMAT 相关表达式，具体格式如下：
+  * DATE_FORMAT(NOW(),'%Y-%m')
+  * DATE_FORMAT(NOW(),'%Y-%m-%d')
+  * DATE_FORMAT(NOW(),'%Y-%m-%d %H.%i.%s')
+  * DATE_FORMAT(NOW(),'%Y-%m-%d %H:%i:%s')
+
+* STR_TO_DATE('1980-01-01','%Y-%m-%d')
```
Comment (hfxsd) [inline data-type-default-values.md:56]:
```suggestion
* `UPPER(SUBSTRING_INDEX(user(),'@',1))`

* `REPLACE(UPPER(uuid()), '-', '')`

* `DATE_FORMAT` 相关表达式，具体格式如下：

  * `DATE_FORMAT(NOW(),'%Y-%m')`
  * `DATE_FORMAT(NOW(),'%Y-%m-%d')`
  * `DATE_FORMAT(NOW(),'%Y-%m-%d %H.%i.%s')`
  * `DATE_FORMAT(NOW(),'%Y-%m-%d %H:%i:%s')`

* `STR_TO_DATE('1980-01-01','%Y-%m-%d')`
```

File: data-type-default-values.md (hunk 1 above)
Comment (zimulala) [reply data-type-default-values.md:44]: @CbcWestwolf 帮忙确认一下，这几个函数是否添加到这里。

File: data-type-default-values.md [hunk 8]
```diff
@@ -24,3 +24,40 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+
+## 表达式默认值
+
+> **警告：**
+>
+> 该特性为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+MySQL 从 8.0.13 开始支持在 `DEFAULT` 子句中指定表达式为默认值。具体可参考 [Explicit Default Handling as of MySQL 8.0.13](https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html#data-type-defaults-explicit)。TiDB 参考了该功能，在 `DEFAULT` 子句中支持指定部分表达式作为字段的默认值。
+
+从 v8.0.0 版本开始，`BLOB`、`TEXT` 以及 `JSON` 数据类型支持为其分配默认值，但是默认值仅支持通过表达式来设置。
```
Comment (hfxsd) [inline data-type-default-values.md:36]:
```suggestion
从 TiDB v8.0.0 开始，`BLOB`、`TEXT` 以及 `JSON` 数据类型支持为其分配默认值，但是默认值仅支持通过表达式来设置。
```

File: data-type-default-values.md [hunk 9]
```diff
@@ -24,3 +24,40 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+* `UPPER(SUBSTRING_INDEX(user(),'@',1))`
+
+* `REPLACE(UPPER(uuid()), '-', '')`
+
+* `DATE_FORMAT` 相关表达式，具体格式如下：
+
+  * `DATE_FORMAT(NOW(),'%Y-%m')`
+  * `DATE_FORMAT(NOW(),'%Y-%m-%d')`
+  * `DATE_FORMAT(NOW(),'%Y-%m-%d %H.%i.%s')`
+  * `DATE_FORMAT(NOW(),'%Y-%m-%d %H:%i:%s')`
```
Comment (hfxsd) [inline data-type-default-values.md:57]:
```suggestion
    * `DATE_FORMAT(NOW(),'%Y-%m')`
    * `DATE_FORMAT(NOW(),'%Y-%m-%d')`
    * `DATE_FORMAT(NOW(),'%Y-%m-%d %H.%i.%s')`
    * `DATE_FORMAT(NOW(),'%Y-%m-%d %H:%i:%s')`
```

File: data-type-default-values.md [hunk 10]
```diff
@@ -24,3 +24,40 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+    * `DATE_FORMAT(NOW(),'%Y-%m')`
+    * `DATE_FORMAT(NOW(),'%Y-%m-%d')`
+    * `DATE_FORMAT(NOW(),'%Y-%m-%d %H.%i.%s')`
+    * `DATE_FORMAT(NOW(),'%Y-%m-%d %H:%i:%s')`
+
+* `STR_TO_DATE('1980-01-01','%Y-%m-%d')`
+
+> **注意：**
+>
+> 目前 `CHANGE COLUMN` 和 `MODIFY COLUMN` 语句不支持使用表达式来设置默认值。
```
Comment (zimulala) [inline data-type-default-values.md:63]:
```suggestion
> 目前 `ADD COLUMN` 语句不支持使用表达式来设置默认值。
```

File: data-type-default-values.md [hunk 11]
```diff
@@ -24,3 +24,40 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+CREATE TABLE t2 (b BLOB DEFAULT (rand()));
+```
+
+TiDB 目前开始支持一些表达式，具体如下：
+
+* RAND, UUID, UUID_TO_BIN
+
+从 v8.0.0 版本开始，`DEFAULT` 子句中支持用以下表达式来设置默认值。
+
+* `UPPER(SUBSTRING_INDEX(user(),'@',1))`
```
Comment (tangenta) [inline data-type-default-values.md:48]:
```suggestion
* `UPPER(SUBSTRING_INDEX(user(), '@', 1))`
```

File: data-type-default-values.md [hunk 12]
```diff
@@ -24,3 +24,40 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+* `REPLACE(UPPER(uuid()), '-', '')`
+
+* `DATE_FORMAT` 相关表达式，具体格式如下：
+
+    * `DATE_FORMAT(NOW(),'%Y-%m')`
+    * `DATE_FORMAT(NOW(),'%Y-%m-%d')`
+    * `DATE_FORMAT(NOW(),'%Y-%m-%d %H.%i.%s')`
+    * `DATE_FORMAT(NOW(),'%Y-%m-%d %H:%i:%s')`
+
+* `STR_TO_DATE('1980-01-01','%Y-%m-%d')`
```
Comment (tangenta) [inline data-type-default-values.md:59]:
```suggestion
    * `DATE_FORMAT(NOW(), '%Y-%m')`
    * `DATE_FORMAT(NOW(), '%Y-%m-%d')`
    * `DATE_FORMAT(NOW(), '%Y-%m-%d %H.%i.%s')`
    * `DATE_FORMAT(NOW(), '%Y-%m-%d %H:%i:%s')`

* `STR_TO_DATE('1980-01-01', '%Y-%m-%d')`
```

**Review by tangenta — APPROVED** [review]
rest LGTM

File: data-type-default-values.md (hunk 1 above)
Comment (CbcWestwolf) [reply data-type-default-values.md:44]: 嗯嗯，放这吧。这几个之前也测试过

**Review by CbcWestwolf — APPROVED** [review]

File: data-type-default-values.md (hunk 1 above)
Comment (zimulala) [reply data-type-default-values.md:44]: 好的~

File: data-type-default-values.md [hunk 13]
```diff
@@ -24,3 +24,40 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+
+从 TiDB v8.0.0 开始，`BLOB`、`TEXT` 以及 `JSON` 数据类型支持为其分配默认值，但是默认值仅支持通过表达式来设置。
+
+```sql
+CREATE TABLE t2 (b BLOB DEFAULT (rand()));
+```
+
+TiDB 目前开始支持一些表达式，具体如下：
+
+* RAND, UUID, UUID_TO_BIN
```
Comment (hfxsd) [inline data-type-default-values.md:44]:
```suggestion
* `RAND`、`UUID`、`UUID_TO_BIN`
```

**Review by hfxsd — APPROVED** [review]

File: data-type-default-values.md [hunk 14]
```diff
@@ -24,3 +24,40 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+
+## 表达式默认值
+
+> **警告：**
+>
+> 该特性为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+MySQL 从 8.0.13 开始支持在 `DEFAULT` 子句中指定表达式为默认值。具体可参考 [Explicit Default Handling as of MySQL 8.0.13](https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html#data-type-defaults-explicit)。TiDB 参考了该功能，在 `DEFAULT` 子句中支持指定部分表达式作为字段的默认值。
+
+从 TiDB v8.0.0 开始，`BLOB`、`TEXT` 以及 `JSON` 数据类型支持为其分配默认值，但是默认值仅支持通过表达式来设置。
```
Comment (hfxsd) [inline data-type-default-values.md:36]:
```suggestion
从 TiDB v8.0.0 开始， [`BLOB`、`TEXT`](/data-type-string.md) 以及 [`JSON`](/data-type-json.md) 数据类型支持为其分配默认值，但是默认值仅支持通过表达式来设置。
```

File: data-type-default-values.md [hunk 15]
```diff
@@ -24,3 +24,40 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+
+```sql
+CREATE TABLE t2 (b BLOB DEFAULT (rand()));
```
Comment (hfxsd) [inline data-type-default-values.md:39]:
```suggestion
CREATE TABLE t2 (b BLOB DEFAULT (RAND()));
```

File: data-type-default-values.md [hunk 16]
```diff
@@ -24,3 +24,40 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+```
+
+TiDB 目前开始支持一些表达式，具体如下：
+
+* `RAND`、`UUID`、`UUID_TO_BIN`
+
+从 v8.0.0 版本开始，`DEFAULT` 子句中支持用以下表达式来设置默认值。
+
+* `UPPER(SUBSTRING_INDEX(user(), '@', 1))`
```
Comment (hfxsd) [inline data-type-default-values.md:48]:
```suggestion
* `UPPER(SUBSTRING_INDEX(USER(), '@', 1))`
```

File: data-type-default-values.md [hunk 17]
```diff
@@ -24,3 +24,40 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+
+* `REPLACE(UPPER(uuid()), '-', '')`
```
Comment (hfxsd) [inline data-type-default-values.md:50]:
```suggestion
* `REPLACE(UPPER(UUID()), '-', '')`
```

File: data-type-default-values.md [hunk 18]
```diff
@@ -24,3 +24,40 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+
+从 TiDB v8.0.0 开始，`BLOB`、`TEXT` 以及 `JSON` 数据类型支持为其分配默认值，但是默认值仅支持通过表达式来设置。
+
+```sql
+CREATE TABLE t2 (b BLOB DEFAULT (rand()));
+```
+
+TiDB 目前开始支持一些表达式，具体如下：
+
+* `RAND`、`UUID`、`UUID_TO_BIN`
```
Comment (hfxsd) [inline data-type-default-values.md:44]:
```suggestion
* [`RAND()`](/functions-and-operators/numeric-functions-and-operators.md)、[`UUID()`](/functions-and-operators/miscellaneous-functions.md)、[`UUID_TO_BIN()`](/functions-and-operators/miscellaneous-functions.md)
```

File: data-type-default-values.md [hunk 19]
```diff
@@ -7,7 +7,7 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
 
 在一个数据类型描述中的 `DEFAULT value` 段描述了一个列的默认值。这个默认值必须是常量，不可以是一个函数或者是表达式。但是对于时间类型，可以例外的使用 `NOW`、`CURRENT_TIMESTAMP`、`LOCALTIME`、`LOCALTIMESTAMP` 等函数作为 `DATETIME` 或者 `TIMESTAMP` 的默认值。
 
-`BLOB`、`TEXT` 以及 `JSON` 不可以设置默认值。
+`BLOB`、`TEXT` 以及 `JSON` 也可以设置默认值。
```
Comment (Oreoxmt) [inline data-type-default-values.md:10]:
```suggestion
从 v8.0.0 开始，[`BLOB`](/data-type-string.md#blob-类型)、[`TEXT`](/data-type-string.md#text-类型) 以及 [`JSON`](/data-type-json.md#json-类型) 可以设置[表达式默认值](#表达式默认值)。
```

**Review by Oreoxmt — APPROVED** [review]

File: data-type-default-values.md [hunk 20]
```diff
@@ -24,3 +24,40 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+
+TiDB 参考了该功能，支持在 `DEFAULT` 子句中指定部分表达式作为字段的默认值。从 v8.0.0 开始，TiDB 支持为 `BLOB`、`TEXT` 以及 `JSON` 数据类型分配默认值，但是默认值仅支持通过表达式来设置。以下是 `BLOB` 的示例：
+
+```sql
+CREATE TABLE t2 (b BLOB DEFAULT (RAND()));
+```
+
+TiDB 目前开始支持一些表达式，具体如下：
+
+* [`RAND()`](/functions-and-operators/numeric-functions-and-operators.md)、[`UUID()`](/functions-and-operators/miscellaneous-functions.md)、[`UUID_TO_BIN()`](/functions-and-operators/miscellaneous-functions.md)
```
Comment (hfxsd) [inline data-type-default-values.md:44]:
```suggestion
* [`RAND()`](/functions-and-operators/numeric-functions-and-operators.md)
* [`UUID()`](/functions-and-operators/miscellaneous-functions.md)
* [`UUID_TO_BIN()`](/functions-and-operators/miscellaneous-functions.md)
```

File: data-type-default-values.md [hunk 21]
```diff
@@ -24,3 +24,42 @@ aliases: ['/docs-cn/dev/data-type-default-values/','/docs-cn/dev/reference/sql/d
...
+
+MySQL 从 8.0.13 开始支持在 `DEFAULT` 子句中指定表达式为默认值。具体可参考 [Explicit Default Handling as of MySQL 8.0.13](https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html#data-type-defaults-explicit)。
+
+TiDB 参考了该功能，支持在 `DEFAULT` 子句中指定部分表达式作为字段的默认值。从 v8.0.0 开始，TiDB 支持为 `BLOB`、`TEXT` 以及 `JSON` 数据类型分配默认值，但是默认值仅支持通过表达式来设置。以下是 `BLOB` 的示例：
+
+```sql
+CREATE TABLE t2 (b BLOB DEFAULT (RAND()));
+```
+
+TiDB 目前开始支持一些表达式，具体如下：
```
Comment (hfxsd) [inline data-type-default-values.md:42]:
```suggestion
TiDB 支持以下表达式：
```

**Pull request merged**